### DISCUSSION TITLE
uvalue and polymorphic types

### DISCUSSION THREAD

**Discussion by innochenti** [BlackMATov/meta.hpp#28] [Q&A] [answered]
Is my understanding correct that uvalue::try_cast should work in case of casting the derived value to the base?

```
struct Base { virtual ~Base() {} }; struct Derived : Base {};

...

meta::uvalue value{new Derived()};
auto base = value.try_cast<Base>(); // currently returns nullptr
```

**Comment by BlackMATov** [comment]
Right now, "uvalue" doesn't support any casts and works like std::any with little specific features. But it can be added because I already do something like this in function calls (when we call functions with reference or pointer attributes on base classes). Of course, it will work only for registered classes with correct "base_<>" attributes.

I'll create a task for this feature!

**Comment by innochenti** [comment]
thank you. I think it will be useful.

**Comment by BlackMATov** [comment, marked as answer]
Upcasting is now supported. New example: [UValue](https://github.com/BlackMATov/meta.hpp/blob/23f21787edc2e694c4dd53a235fd48ba571ae288/manuals/meta_examples/uvalue_example.cpp)
